tools/fastlane-plugin/lib/fastlane/plugin/bugsnag/actions/send_build_to_bugsnag.rb

```diff
@@ -1,63 +1,98 @@
 require "xmlsimple"
 require "json"
 require_relative "find_info_plist_path"
+require_relative "bugsnag_cli"
 
 module Fastlane
   module Actions
     class SendBuildToBugsnagAction < Action
-
-      BUILD_TOOL = "bugsnag-fastlane-plugin"
-
       def self.run(params)
-        payload = {buildTool: BUILD_TOOL, sourceControl: {}}
+        bugsnag_cli_path = BugsnagCli.get_bugsnag_cli_path(params)
+        verbose = UI.verbose("Using bugsnag-cli from path: #{bugsnag_cli_path}")
 
         # If a configuration file was found or was specified, load in the options:
+        config_options = {}
         if params[:config_file]
           UI.message("Loading build information from #{params[:config_file]}")
           config_options = load_config_file_options(params[:config_file])
-
-          # for each of the config options, if it's not been overriden by any
-          # input to the lane, write it to the payload:
-          payload[:apiKey] = params[:api_key] || config_options[:apiKey]
-          payload[:appVersion] = params[:app_version] || config_options[:appVersion]
-          payload[:appVersionCode] = params[:android_version_code] || config_options[:appVersionCode]
-          payload[:appBundleVersion] = params[:ios_bundle_version] || config_options[:appBundleVersion]
-          payload[:releaseStage] = params[:release_stage] || config_options[:releaseStage] || "production"
-        else
-          # No configuration file was found or specified, use the input parameters:
-          payload[:apiKey] = params[:api_key]
-          payload[:appVersion] = params[:app_version]
-          payload[:appVersionCode] = params[:android_version_code]
-          payload[:appBundleVersion] = params[:ios_bundle_version]
-          payload[:releaseStage] = params[:release_stage] || "production"
         end
 
-        # If builder, or source control information has been provided into
-        # Fastlane, apply it to the payload here.
-        payload[:builderName] = params[:builder] if params[:builder]
-        payload[:sourceControl][:revision] = params[:revision] if params[:revision]
-        payload[:sourceControl][:repository] = params[:repository] if params[:repository]
-        payload[:sourceControl][:provider] = params[:provider] if params[:provider]
-
-        # If provided apply metadata to payload.
-        payload[:metadata] = params[:metadata]
-
-        payload.reject! {|k,v| v == nil || (v.is_a?(Hash) && v.empty?)}
-
-        if payload[:apiKey].nil? || !payload[:apiKey].is_a?(String)
+        api_key = params[:api_key] || config_options[:apiKey] unless (params[:api_key].nil? && config_options[:apiKey].nil?)
```

Comment:
This might be an existing issue (or perhaps I've missed it), but we don't read plist info for API key and version fields and probably we should.

Reply:
Discussed IRL